storage/storage_dest.go

```diff
@@ -445,16 +454,42 @@ func (s *storageImageDestination) computeID(m manifest.Manifest) string {
 			}
 			diffIDs = append([]digest.Digest{diffID}, diffIDs...)
 		}
-	case *manifest.Schema2, *manifest.OCI1:
-		// We know the ID calculation for these formats doesn't actually use the diffIDs,
-		// so we don't need to populate the diffID list.
+	case *manifest.Schema2:
+		// We know the ID calculation doesn't actually use the diffIDs, so we don't need to populate
+		// the diffID list.
+	case *manifest.OCI1:
+		for _, layer := range m.Layers {
+			toc, err := chunked.GetTOCDigest(layer.Annotations)
+			if err != nil {
+				logrus.Infof("error looking up annotation for layer %q: %v", layer.Digest, err)
+				continue
```

Comment:
Why is it OK to silently continue here?

---

OTOH this _always_ fails on non-Linux. Is that right? Do we want the same image to have different IDs on different platforms?

That might well be right … but… it seems non-trivial. What is the consistent model here?

One possibility is to make `GetTOCDigest` OS-neutral (and move it into a lighter-weight subpackage); then we might need to have some other place to decide whether the OS supports chunked.

Reply:
With #2822, image IDs are decided based on how the layer was actually pulled.